ams/optimizer/src/main/java/com/netease/arctic/optimizer/flink/FlinkOptimizer.java

```diff
@@ -36,4 +44,25 @@ public static void main(String[] args) throws CmdLineException {
       LOG.error("Execute flink optimizer failed", e);
     }
   }
+
+  private static void calcOptimizerMemory(OptimizerConfig config, StreamExecutionEnvironment env) {
+    ReadableConfig configuration = env.getConfiguration();
+    MemorySize jobMemorySize = configuration.get(JobManagerOptions.TOTAL_PROCESS_MEMORY);
+    MemorySize taskMemorySize = configuration.get(TaskManagerOptions.TOTAL_PROCESS_MEMORY);
+    if (jobMemorySize == null || taskMemorySize == null) {
+      // Running locally will not get the configuration.
+      return;
+    }
+    int parallelism = config.getExecutionParallel();
+    int numberOfTaskSlots = configuration.get(TaskManagerOptions.NUM_TASK_SLOTS);
+    int memorySize = jobMemorySize.getMebiBytes() + (parallelism / numberOfTaskSlots) * taskMemorySize.getMebiBytes();
```

Comment:
The TaskManager's count cannot be simply calculated using parallelism/numberOfTaskSlots. The capacity of the TaskManager is related to many configuration parameters in Flink. Can the calculation method used in Flink be reused here?

Reply:
In the following code, when calculating the TaskManager's count, it takes into account rounding up the result of `parallelism/numberOfTaskSlots`.

Can you provide more details of mechanisms which may affect the count of TaskManagers? @shidayang

Reply:
The Flink code has been changed significantly and I couldn't find it，so this calculation method is also okay.